src/ILLink.Shared/TrimAnalysis/RequireDynamicallyAccessedMembersAction.cs

```diff
@@ -34,6 +34,8 @@ public void Invoke (in DiagnosticContext diagnosticContext, in MultiValue value,
 					}
 				} else if (uniqueValue == NullValue.Instance) {
 					// Ignore - probably unreachable path as it would fail at runtime anyway.
+				} else if (uniqueValue is ArrayValue) {
+					// Ignore 
```

Comment:
Arrays in this case should be handled as "unknown" or "unsupported" and fall through to the else case below. It means that either data flow screwed up, or that there something really weird going on in the code.
Technically it should not be possible to get an ArrayValue here, the source/target for this method should be statically typed as either `Sytem.Type` or `System.String` and nothing else (since we can't/don't handle annotations on anything else).